### PR TITLE
fix(semantic hub): display error bar

### PR DESCRIPTION
**Pull request by lavanya-bmw**
## Description

Show appropriate error information to the user along with reload

## Why

To enhance error handling

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/891

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/pages/SemanticHub/ModelTable.tsx [hunk 1]
```diff
@@ -149,6 +149,18 @@
...
   const columns = SemanticModelTableColumns(t, onModelSelect)
+  const errorObj = {
+    status: 0,
+    message: '',
+  }
+
+  if (error) {
+    errorObj.status = Number(error)
+    errorObj.message =
+      error && Number(error) >= 400 && Number(error) < 500
```
[the remark this answers is not in the document]
Comment (lavanya-bmw) [reply src/components/pages/SemanticHub/ModelTable.tsx:160]: Fixed

**Review by manojava-gk — APPROVED** [review]
lgtm

File: src/features/semanticModels/types.ts [hunk 2]
```diff
@@ -30,7 +30,9 @@ export interface SemanticModelsInitialState {
   uploadError: string
   openApiLink: string
   openApiError: string
-  error: string
+  // Add an ESLint exception until there is a solution
+  // eslint-disable-next-line
+  error: any
```
Comment (oyo) [inline src/features/semanticModels/types.ts:35]: please define an interface for ErrorResponse containing all the attributes you like to access. Like here we use `error.response.status` 

```
export interface ErrorResponse {
  response: {
    status: number
  }
}

...

  error: ErrorResponse
```

File: src/features/semanticModels/actions.ts [hunk 3]
```diff
@@ -29,9 +29,11 @@ const fetchSemanticModels = createAsyncThunk(
   async ({ filter }: { filter: FilterParams }) => {
     try {
       return await Api.getInstance().getModels(filter)
-    } catch (error: unknown) {
+      // Add an ESLint exception until there is a solution
+      // eslint-disable-next-line
+    } catch (error: any) {
```
Comment (oyo) [inline src/features/semanticModels/actions.ts:34]: import and use the ErrorResponse type from other comment

**Review by oyo — CHANGES_REQUESTED** [review]
see comments about how to get rid of any type

File: src/features/semanticModels/actions.ts [hunk 4]
```diff
@@ -29,9 +29,11 @@ const fetchSemanticModels = createAsyncThunk(
...
       console.error('api call error:', error)
-      throw Error(`Fetching models failed: ${message}`)
+      throw Error(JSON.stringify(error.response.status))
```
Comment (oyo) [inline src/features/semanticModels/actions.ts:36]: `(error as ErrorResponse).response.status`

**Review by oyo — CHANGES_REQUESTED** [review]

**Review by oyo — APPROVED** [review]

**Pull request merged**